src/plots/cartesian/select.js

```diff
@@ -349,11 +323,246 @@ function prepSelect(e, startX, startY, dragOptions, mode) {
+        }
+    }
+
+    function isHoverDataSet(hoverData) {
+        return hoverData &&
+          Array.isArray(hoverData) &&
+          hoverData[0].hoverOnBox !== true;
+    }
+
+    /**
+     * Function to determine the clicked points for the given searchInfo (trace)
+     * based on the passed hoverData.
+     *
+     * Function assumes the following about hoverData:
+     * - when hoverData has more than one element (e.g. box trace),
+     *   if a point is hovered upon, the clicked point is the first
+     *   element in the array. It is assumed that fx/hover.js and satellite
+     *   modules are doing that correctly.
+     * - at the moment only one point at a time is considered to be selected
+     *   upon one click.
+     *
+     * Function also encapsulates special knowledge about the slight
+     * inconsistencies in what hoverData can look like for different
+     * trace types. As hoverData will become more homogeneous, this
+     * logic will become cleaner.
+     *
+     * See https://github.com/plotly/plotly.js/issues/1852 for the
+     * respective discussion.
+     */
+    function clickedPtsFor(searchInfo, hoverData) {
+        var clickedPts = [];
+        var hoverDatum;
+
+        if(hoverData.length > 0) {
+            hoverDatum = hoverData[0];
+            if(hoverDatum.fullData._expandedIndex === searchInfo.cd[0].trace._expandedIndex) {
+                // Special case for box (and violin)
+                if(hoverDatum.hoverOnBox === true) return clickedPts;
+
+                // TODO hoverDatum not having a pointNumber but a binNumber seems to be an oddity of histogram only
+                // Not deleting .pointNumber in histogram/event_data.js would simplify code here and in addition
+                // would not break the hover event structure
+                // documented at https://plot.ly/javascript/hover-events/
+                if(hoverDatum.pointNumber !== undefined) {
+                    clickedPts.push({
+                        pointNumber: hoverDatum.pointNumber
+                    });
+                } else if(hoverDatum.binNumber !== undefined) {
+                    clickedPts.push({
+                        pointNumber: hoverDatum.binNumber,
+                        pointNumbers: hoverDatum.pointNumbers
+                    });
+                }
+            }
+        }
+
+        return clickedPts;
+    }
+}
+
+function determineSearchTraces(gd, xAxes, yAxes) {
+    var searchTraces = [];
+    var xAxisIds = xAxes.map(getAxId);
+    var yAxisIds = yAxes.map(getAxId);
+    var cd;
+    var trace;
+    var i;
+
+    for(i = 0; i < gd.calcdata.length; i++) {
+        cd = gd.calcdata[i];
+        trace = cd[0].trace;
+
+        if(trace.visible !== true || !trace._module ||
+          !trace._module.toggleSelected || !trace._module.getPointsIn) continue;
+
+        // TODO is dragOptions.subplot is ever set? If not, delete.
+        // if(dragOptions.subplot) {
```

Comment:
Looks like [ternary subplots still set that _subplot_ field](https://github.com/plotly/plotly.js/blob/24b291285a15650a76ae6c78e2762a8e88f2027d/src/plots/ternary/ternary.js#L458) as well as geo and mapbox subplots.

Reply:
Thx! Really great you're also going over my TODOs because some of them would have required your or AJ's input anyways!